src/Enchantments.cpp

```diff
@@ -242,6 +242,323 @@ bool cEnchantments::operator !=(const cEnchantments & a_Other) const
+			{
+				valid = false;
+			}
+		}
+
+		else if (ItemCategory::IsShovel(a_ItemType))
+		{
+			if ((a_EnchantmentID != cEnchantments::enchEfficiency) &&
+				(a_EnchantmentID != cEnchantments::enchSilkTouch) &&
+				(a_EnchantmentID != cEnchantments::enchFortune) &&
+				(a_EnchantmentID != cEnchantments::enchUnbreaking))
+			{
+				valid = false;
+			}
+		}
+
+		else if (ItemCategory::IsPickaxe(a_ItemType))
+		{
+			if ((a_EnchantmentID != cEnchantments::enchEfficiency) &&
+				(a_EnchantmentID != cEnchantments::enchSilkTouch) &&
+				(a_EnchantmentID != cEnchantments::enchFortune) &&
+				(a_EnchantmentID != cEnchantments::enchUnbreaking))
+			{
+				valid = false;
+			}
+		}
+
+		else if (ItemCategory::IsHoe(a_ItemType))
+		{
+			if ((a_EnchantmentID != cEnchantments::enchUnbreaking))
+			{
+				valid = false;
+			}
+		}
+	}
+
+	else if (ItemCategory::IsArmor(a_ItemType))
+	{
+		if (ItemCategory::IsHelmet(a_ItemType))
+		{
+			if ((a_EnchantmentID != cEnchantments::enchProtection) &&
+				(a_EnchantmentID != cEnchantments::enchFireProtection) &&
+				(a_EnchantmentID != cEnchantments::enchBlastProtection) &&
+				(a_EnchantmentID != cEnchantments::enchProjectileProtection) &&
+				(a_EnchantmentID != cEnchantments::enchRespiration) &&
+				(a_EnchantmentID != cEnchantments::enchAquaAffinity) &&
+				(a_EnchantmentID != cEnchantments::enchThorns) &&
+				(a_EnchantmentID != cEnchantments::enchUnbreaking))
+			{
+				valid = false;
+			}
+		}
+
+		else if (ItemCategory::IsChestPlate(a_ItemType))
+		{
+			if ((a_EnchantmentID != cEnchantments::enchProtection) &&
+				(a_EnchantmentID != cEnchantments::enchFireProtection) &&
+				(a_EnchantmentID != cEnchantments::enchBlastProtection) &&
+				(a_EnchantmentID != cEnchantments::enchProjectileProtection) &&
+				(a_EnchantmentID != cEnchantments::enchThorns) &&
+				(a_EnchantmentID != cEnchantments::enchUnbreaking))
+			{
+				valid = false;
+			}
+		}
+
+		else if (ItemCategory::IsLeggings(a_ItemType))
+		{
+			if ((a_EnchantmentID != cEnchantments::enchProtection) &&
+				(a_EnchantmentID != cEnchantments::enchFireProtection) &&
+				(a_EnchantmentID != cEnchantments::enchBlastProtection) &&
+				(a_EnchantmentID != cEnchantments::enchProjectileProtection) &&
+				(a_EnchantmentID != cEnchantments::enchThorns) &&
+				(a_EnchantmentID != cEnchantments::enchUnbreaking))
+			{
+				valid = false;
+			}
+		}
+
+		else if (ItemCategory::IsBoots(a_ItemType))
+		{
+			if ((a_EnchantmentID != cEnchantments::enchProtection) &&
+				(a_EnchantmentID != cEnchantments::enchFireProtection) &&
+				(a_EnchantmentID != cEnchantments::enchFeatherFalling) &&
+				(a_EnchantmentID != cEnchantments::enchBlastProtection) &&
+				(a_EnchantmentID != cEnchantments::enchProjectileProtection) &&
+				(a_EnchantmentID != cEnchantments::enchThorns) &&
+				(a_EnchantmentID != cEnchantments::enchDepthStrider) &&
+				(a_EnchantmentID != cEnchantments::enchUnbreaking))
+			{
+				valid = false;
+			}
+		}
+	}
```

Comment:
I can't think of any exmples in the source off the top of my head, but what I was thinking was std::map<ItemID, std::set<cEnchantments>> or std::map<cEnchantments, std::set<itemID>>. The only problem on reflection is the categories. I suspect that we would need an object that represents a set of items, and contains a mixture of items and categories. I'm not sure std::set<itemID> would cut it.